Source/WebCore/dom/CustomElementRegistry.h

```diff
@@ -61,8 +64,26 @@ class CustomElementRegistry : public RefCounted<CustomElementRegistry>, public C
     static Ref<CustomElementRegistry> create(ScriptExecutionContext&);
     ~CustomElementRegistry();
 
+    bool isScoped() const { return !m_window; }
     Document* document() const;
 
+    static CustomElementRegistry* registryForElement(const Element& element)
+    {
+        if (UNLIKELY(element.usesScopedCustomElementRegistryMap()))
+            return scopedCustomElementRegistryMap().get(element);
+        return element.treeScope().customElementRegistry();
+    }
+
+    static CustomElementRegistry* registryForNodeOrTreeScope(const Node& node, const TreeScope& treeScope)
+    {
+        if (UNLIKELY(is<Element>(node) && node.usesScopedCustomElementRegistryMap()))
```

Comment:
Would be nicer to use dynamicDowncast in new code.

Reply:
Sure, we can do that.